src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -141,6 +144,14 @@ public string GetShortFolderName()
             return GetShortFolderName(DefaultFrameworkNameProvider.Instance);
         }
 
+        /// <summary>
+        /// Helper that is .NET 5 Era aware to replace identifier when appropriate
+        /// </summary>
+        private string GetFrameworkIdentifier()
+        {
+            return IsNet5Era ? FrameworkConstants.FrameworkIdentifiers.Net : Framework;
```

Comment:
This breaks all of our IFrameworkNameProvider logic.
Feels like we should call that out somehow, but I'm not sure how.

Reply:
Actually you can also consider using mappings.GetShortNameReplacement? 

There's already a method to do the rewrites :) 

https://github.com/NuGet/NuGet.Client/blob/ca008ff53cd941d31cba2052c1fcf4c31770b294/src/NuGet.Core/NuGet.Frameworks/FrameworkNameProvider.cs#L1030

https://github.com/NuGet/NuGet.Client/blob/ca008ff53cd941d31cba2052c1fcf4c31770b294/src/NuGet.Core/NuGet.Frameworks/def/IFrameworkNameProvider.cs#L137

Reply:
is there a more elegant way we should do it?
What scenarios are we breaking?

Reply:
will look at that. did some looking at mappings, but didn't see what i needed. may have missed that.

Reply:
I don't think we are breaking any scenarios, we are just breaking the abstraction.

I am ok with either approach.
The requirement is imperfect, but maybe the ShortNameReplacement concept helps.

Reply:
i think it is reasonable to do this investigation and likely change post-preview2, given the short timelines for preview2.
@nkolev92  - thoughts?

Reply:
makes sense.